packages/website/ts/components/relayer_index/relayer_index.tsx

```diff
@@ -25,94 +32,65 @@ const styles: Styles = {
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '5',
-        headerUrl: '/images/og_image.png',
-        name: 'blahg',
-        marketShare: 0.5,
-        topTokens,
-    },
-    {
-        id: '6',
-        headerUrl: '/images/og_image.png',
-        name: 'hello',
-        marketShare: 0.5,
-        topTokens,
-    },
-];
-
 const CELL_HEIGHT = 260;
 const NUMBER_OF_COLUMNS = 4;
 const GRID_PADDING = 16;
 
-export const RelayerIndex: React.StatelessComponent<RelayerIndexProps> = (props: RelayerIndexProps) => {
-    return (
-        <div style={styles.root}>
-            <GridList cellHeight={CELL_HEIGHT} cols={NUMBER_OF_COLUMNS} padding={GRID_PADDING} style={styles.gridList}>
-                {relayerInfos.map((relayerInfo: RelayerInfo) => (
-                    <RelayerGridTile key={relayerInfo.id} relayerInfo={relayerInfo} networkId={props.networkId} />
-                ))}
-            </GridList>
-        </div>
-    );
-};
+export class RelayerIndex extends React.Component<RelayerIndexProps, RelayerIndexState> {
+    private _isUnmounted: boolean;
+    constructor(props: RelayerIndexProps) {
+        super(props);
+        this._isUnmounted = false;
+        this.state = {
+            relayerInfos: undefined,
+            error: undefined,
+        };
+    }
+    public componentWillMount() {
+        // tslint:disable-next-line:no-floating-promises
+        this._fetchRelayerInfosAsync();
+    }
+    public componentWillUnmount() {
+        this._isUnmounted = true;
+    }
+    public render() {
+        // TODO: loading and error states with a scrolling container
+        const readyToRender = _.isUndefined(this.state.error) && !_.isUndefined(this.state.relayerInfos);
+        return (
+            readyToRender && (
+                <div style={styles.root}>
+                    <GridList
+                        cellHeight={CELL_HEIGHT}
+                        cols={NUMBER_OF_COLUMNS}
+                        padding={GRID_PADDING}
+                        style={styles.gridList}
+                    >
+                        {this.state.relayerInfos.map((relayerInfo: WebsiteBackendRelayerInfo) => (
+                            <RelayerGridTile
+                                key={relayerInfo.id}
+                                relayerInfo={relayerInfo}
+                                networkId={this.props.networkId}
+                            />
+                        ))}
+                    </GridList>
+                </div>
+            )
+        );
+    }
+    private async _fetchRelayerInfosAsync(): Promise<void> {
+        try {
+            const relayerInfos = await backendClient.getRelayerInfosAsync();
+            if (!this._isUnmounted) {
+                this.setState({
+                    relayerInfos,
+                });
+            }
+        } catch (error) {
+            if (!this._isUnmounted) {
+                this.setState({
+                    error,
```

Comment:
Do we want to set the entire error message as the state? Maybe just an `errMsg` that could be displayed? My reasoning is that it probably makes more sense to convert Error objects into error messages suitable for this component in one place, rather then in a bunch of places.

Reply:
I think I like keeping the entire error as the state and keeping the display logic in the render method when its time to display to the user